BlogTemplate/Pages/Post.cshtml.cs

```diff
@@ -31,14 +31,21 @@ public HtmlString HtmlBody()
             return html;
         }
 
-        public void OnGet([FromRoute] int id)
+        public IActionResult OnGet([FromRoute] int id, [FromRoute] string slug)
         {
             Post = _dataStore.GetPost(id);
 
             if (Post == null)
             {
-                RedirectToPage("/Index");
+                return RedirectToPage("/Index");
+            }
+
+            if (slug != Post.Slug)
+            {
+                return RedirectPermanent($"/Post/{Post.Id}/{Post.Slug}");

```

Comment:
Shouldn't an error or exception be thrown if the slug is not correct?

Reply:
We just want to redirect because getting the post does not depend on the slug. Also using a redirect helps with when the slug is updated because external links that use the old slug will have a "moved permanently" attached to them so people will not lose SEO on their posts.